### DISCUSSION TITLE
deepspeed ignoring steps_per_print

### DISCUSSION THREAD

**Discussion by ansar-ayan** [deepspeedai/DeepSpeed#6521] [Q&A]
Hi everyone, 
deepspeed is ignoring the step_per_print parameter in logging. is this a bug and is there a workaround to this?

**Comment by tjruwase** [comment]
@ansar-ayan, can you please share more details of the problem you are seeing. Below is a use of the parameter. 
https://github.com/microsoft/DeepSpeed/blob/c27483933d50a693fef9c48418d2664cf6a6a6f8/deepspeed/runtime/engine.py#L2169

**Reply by ansar-ayan** [reply]
Hey @tjruwase, Its when I am passing the config 
 `
  model, _, _, _ = deepspeed.initialize(
            model=model,
            model_parameters=model.parameters(),
            config={
                "train_micro_batch_size_per_gpu": batch_size,
                "optimizer": {
                    "type": "Adam",
                    "params": {"lr": 1e-4},
                },
                "zero_optimization": {
                    "stage": 1,
                    "offload_optimizer": {"device": "cpu"},
                },
                "logging": {
                    "steps_per_print": 10000,
                },
            },
        )
`
Even after specifying the logging, it is ignoring the param. I am using the 0.15.1 version. Will depreciating the version work?

**Reply by ansar-ayan** [reply]
@tjruwase I was incorrectly using step_per_print. 
Its working correctly now. Thanks for your help :)

**Reply by tjruwase** [reply]
@ansar-ayan, I am glad you are unblocked. Sorry that you wasted time on this since ideally DeepSpeed should have alerted you on the typo. We have partially enabled configuration validation for ZeRO via https://github.com/microsoft/DeepSpeed/pull/2004 to detect these kinds of typos. Unfortunately, we currently lack bandwidth to extend to other parts of the codebase. We would appreciate if you had bandwidth to help with this feature. Thanks!
